### PR TITLE
docs: add ruby quickstart documentation

### PR DESCRIPTION
**Pull request by Eazybright**
Document a Ruby quickstart for Novu API

Screenshot 2023-07-06 at 10 07 40 PM
TODO: 

1. This file should be uploaded so it can be added to the docs

2. Add link to sample code - https://github.com/Eazybright/novu-ruby-quickstart.git

### CODE REVIEW

**Comment by unicodeveloper** [comment]
Hey @Eazybright please can you transfer the novu ruby quickstart repo to my account (
@unicodeveloper )? That way, I can transfer it to the Novu org.

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+Or add it to your Gemfile:
+
+```ruby
+gem 'novu'
+```
+
+Then run `bundle install`.
+
+## Initialisation and Usage
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:46]: => Initialize Novu Ruby SDK

Comment (unicodeveloper) [reply docs/docs/overview/quickstart/get-started-with-ruby.md:46]: => Initialize & Configure the Novu Ruby SDK

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+To use the library, Create a new file `index.rb` then initialize the client with your API token:
+
+```ruby
+require 'novu'
+
+client = Novu::Client.new('MY_API_TOKEN')
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:53]: MY_API_TOKEN => YOUR_NOVU_API_KEY

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+```
+
+Replace the `MY_API_TOKEN` argument with the authentic key from the **API Key** section of your [Novu Dashboard](https://web.novu.co/settings).
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:56]: => YOUR_NOVU_API_KEY

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+| Channel | Providers                                                           |
+| ------- | ------------------------------------------------------------------- |
+| Email   | MailGun, Mandrill, MailJet, Amazon SES, Sendgrid, Postmark, Netcore |
+| SMS     | Twilio, Amazon SNS, Plivo, SMS, SMSCentral, Kannel, Infobip, Termii |
+| Chat    | Mattermost, Slack, Microsoft Teams, Discord                         |
+| Push    | FCM, APNS, Expo                                                     |
+
+Only one provider can be **active** per **channel**. Connect any of your favorite providers to get started. The email channel comes with Novu's email provider, which is active by default and includes 300 credits.
+
+## Create A Notification Template
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:96]: Create A Notification Template => Create a Workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+A notification template is the blueprint for the notifications that will be sent. It holds the entire flow of messages sent to the subscriber. This is where all the different channels are tied together under a single entity.
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:98]: notification template => notification workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+The template includes the following:
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:100]: template => workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+
+- Notification template name and Identifier
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:102]: => Notification workflow name and Identifier

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+|         | 2. Click and place UI items with the visual template editor.                                  |
+| SMS     | Text with the option to use handlebars syntax, {{ }} to inject custom variables.              |
+| Chat    | Text with the option to use handlebars syntax, {{ }} to inject custom variables.              |
+| In-App  | Text                                                                                          |
+
+:::info
+Proper authorization needs to be set for the Chat channel for subscribers.
+:::
+
+Please proceed to create a notification template.
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:117]: notification template => notification workflow

**Comment by Eazybright** [comment]
> unicodeveloper

Transfer done.

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+
+1. Click **Workflows** on the left sidebar of your Novu dashboard. 
+TODO: (Add a picture that shows the sidebar)
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:120]: When are you doing to add the TODO:?

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+2. Click the **Create Workflow** button on the top right.
+3. The name of a new notification template is currently **Untitled.** Rename it to a more suitable title.
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:122]: notification template => notification workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+4. Select **Email** as the channel you want to add.
+   ![Select Email as Channel](https://res.cloudinary.com/dxc6bnman/image/upload/f_auto,q_auto/v1685466071/guides/set-email_1_aisoz4.png)
+5. Click on the recently added channel, fill the email subject and click **Update**.
+   ![Update](https://res.cloudinary.com/dxc6bnman/image/upload/f_auto,q_auto/v1685466074/guides/update_email_template_1_exxybg.png)
+6. Click on the **Test** tab and send a test email to verify your notification template.
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:127]: notification template => notification workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+    'to' => {
+        'subscriberId' => '7789'
+    }
+}
+client.trigger_event(trigger_payload)
+```
+
+Before running the code, make sure you understand the following:
+
+- The value of `name` should be the notification template’s trigger ID/slug.
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:206]: notification template => notification workflow

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+
+```ruby
+topicKey = 'frontend-users'
+subscribers = ['6460925ce1a93324257d2fc1', '7789'].to_s
+
+# Add subscribers to a topic
+client.add_subscribers(topicKey, subscribers)
+```
+
+Also, you can remove subscribers from a topic by using the coce snippet below:
```
Comment (unicodeveloper) [inline docs/docs/overview/quickstart/get-started-with-ruby.md:264]: coce => code

**Review by unicodeveloper — COMMENTED** [review]
Great work! Please attend to the comments. Thank you 🙏🏿

File: docs/docs/overview/quickstart/get-started-with-ruby.md (hunk 9 above)
Comment (Eazybright) [reply docs/docs/overview/quickstart/get-started-with-ruby.md:120]: I attached the picture to the description of this PR. 

I noticed that the images are served from cloudinary. i was hoping you help with the upload and update the part with the correct image url

File: docs/docs/overview/quickstart/get-started-with-ruby.md [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+- Send notifications via topics.
+
+## Requirements
+
+To follow the steps in this quickstart, you'll need:
+
+- A Novu account. [Sign up for free](http://web.novu.co) if you don’t have one yet.
+- A working Ruby development environment with at least version 2.6.0.
+
+You can also [view the completed code](https://github.com/novuhq/novu-php-quickstart) of this quick start in a GitHub repo.
```
[the remark this answers is not in the document]
Comment (Eazybright) [reply docs/docs/overview/quickstart/get-started-with-ruby.md:26]: Transfer done.

**Comment by unicodeveloper** [comment]
@Eazybright This is the quickstart URL now https://github.com/novuhq/novu-ruby-quickstart. Please update the guide with it.

**Comment by Eazybright** [comment]
> Great work! Please attend to the comments. Thank you 🙏🏿

PR updated

File: docs/docs/overview/quickstart/get-started-with-ruby.md (hunk 9 above)
Comment (unicodeveloper) [reply docs/docs/overview/quickstart/get-started-with-ruby.md:120]: Oh okay. This is the URL: https://res.cloudinary.com/dxc6bnman/image/upload/v1688745999/guides/workflows_m60hkc.png

Comment (Eazybright) [reply docs/docs/overview/quickstart/get-started-with-ruby.md]: done

**Review by unicodeveloper — APPROVED** [review]

**Comment by unicodeveloper** [comment]
LGTM ⭐  🚀 

Thank you & well done! @Eazybright

**Pull request merged**